File: src/FormingVectorModel/FormingVectorModel/Core/IO/Reader.cpp
#include "Reader.h"
#include "opencv2/opencv.hpp"
#include "../Objects/Visual/Image.h"

Core::Objects::Visual::IImage* Core::IO::Reader::read(std::string path)
{
	cv::Mat mat = cv::imread(path);
	Objects::Visual::IImage* image = new Objects::Visual::Image(mat.rows, mat.cols);
	cv::Vec3b row_color;

	for (int i = 0; i < mat.rows; i++)
	{
		for (int j = 0; j < mat.cols; j++)
		{
			row_color = mat.at<cv::Vec3b>(i, j);
			Objects::Visual::Color color(row_color.val[2], row_color.val[1], row_color.val[0]);
			image->set(i, j, &color);
		}
	}

	return image;
}

Core::Objects::Visual::IImage* Core::IO::Reader::read(Path path)
{
	return read(path.full_path());
}

File: src/FormingVectorModel/FormingVectorModelLibrary/Core/IO/Reader.cs
using System;
using System.IO;
using System.Runtime.InteropServices;
using FormingVectorModelLibrary.Core.Objects.Visual;
using Emgu.CV;
using Emgu.CV.Structure;

namespace FormingVectorModelLibrary.Core.IO
{
    public static class Reader
    {
        public static Objects.Visual.IImage Read(string path)
        {
            Image<Bgr, Byte> rowImage = new Image<Bgr, Byte>(path);
            Objects.Visual.IImage image = new Objects.Visual.Image(rowImage.Rows, rowImage.Cols, rowImage.Data);
            return image;
        }
    }
}